python/experimental/options_search/my_utils.py

```diff
@@ -0,0 +1,212 @@
+        mean_time = np.mean(tc_time_list)
+        return mean_time
+    elif(estimator == "median"):
+        median_time = np.median(tc_time_list)
+        return median_time
+    elif(estimator == "p25"):
+        p25_time = np.percentile(tc_time_list, 25)
+        return p25_time
+    print("Unknown estimator")
+    return infty
+
+def getRawVectorFromTcOpt(tc_opt):
+    tr_dic = {"Max":0, "Preserve3Coincident":1, "Min":2}
+    opt_vect = np.zeros(NB_HYPERPARAMS).astype(int)
+    opt_vect[0] = tr_dic[tc_opt["outerScheduleFusionStrategy"]]
+    opt_vect[1] = tr_dic[tc_opt["intraTileScheduleFusionStrategy"]]
+    opt_vect[2] = tc_opt["fixParametersBeforeScheduling"]
+    opt_vect[3] = len(tc_opt["tile"])
+    opt_vect[4:4+opt_vect[3]] = tc_opt["tile"]
+    opt_vect[10] = tc_opt["unroll"]
+    #opt_vect[11] = tc_opt["tileImperfectlyNested"] #todo: pybind
+    opt_vect[11] = tc_opt["matchLibraryCalls"]
+    opt_vect[12] = len(tc_opt["mapToBlocks"])
+    opt_vect[13:13+opt_vect[12]] = tc_opt["mapToBlocks"]
+    opt_vect[16] = len(tc_opt["mapToThreads"])
+    opt_vect[17:17+opt_vect[16]] = tc_opt["mapToThreads"]
+    opt_vect[20] = tc_opt["useSharedMemory"]
+    opt_vect[21] = tc_opt["usePrivateMemory"]
+    opt_vect[22] = tc_opt["unrollCopyShared"]
+    #opt_vect[23] = tc_opt["maxSharedMemory"]
+    opt_vect[24] = tc_opt["useReadOnlyCache"]
+    opt_vect[25] = tc_opt["privateDepth"]
+    return opt_vect
+
+def optionsFromVector(vect):
+    strat_str = ["Max", "Preserve3Coincident", "Min"]
+    options = tc.MappingOptions("naive")
+    options.outerScheduleFusionStrategy(strat_str[vect[0]])
+    options.intraTileScheduleFusionStrategy(strat_str[vect[1]])
+    options.fixParametersBeforeScheduling(vect[2])
+    options.tile(list(vect[4:(4+vect[3])]))
+    options.unroll(vect[10])
+    options.matchLibraryCalls(vect[11])
+    options.mapToBlocks(list(vect[13:13+vect[12]]))
+    options.mapToThreads(list(vect[17:17+vect[16]])) #grid?
+    options.useSharedMemory(vect[20])
+    options.usePrivateMemory(vect[21])
+    options.unrollCopyShared(vect[22])
+    #options.maxSharedMemory(vect[23])
+    options.useReadOnlyCache(vect[24])
+    options.privateDepth(vect[25])
+    return options
+
+def computeDivs(sz):
+    l = []
+    for i in range(sz): #or 10?
+        if(2**i > sz):
+            break
+        l.append((sz+2**i-1)//(2**i))
+    return l
+
+def getAllDivs(inp, maxp2=8):
+    p2 = []
+    pp=1
+    for i in range(maxp2+1):
+        p2.append(pp)
+        pp*=2
+    l = []
+    for elem in inp:
+        for sz in elem.shape:
+            l += computeDivs(sz)
+    my_list = list(set(l+p2))
+    return np.sort(my_list).tolist()
+
+def computeCat(inp_arg):
+    global cat_sz, cat_val, inp
+    inp = inp_arg
+    cat_sz = np.zeros(NB_HYPERPARAMS).astype(int)
+    cat_val = []
+
+    divs = getAllDivs(inp)
+    #divs2 = getAllDivs([np.array([tc.tclib.shared_memory_size()])])
+
+    cat_val.append([0,1,2])                    #0
```

Comment:
I suppose the comments in this function relate the index of the respective value in the vector of tunable parameters.  Would be nice to have at least a comment about that.

More generally, I would suggest having a bi-directional mapping between indices and names in one place and just using it everywhere else.

I find this much more readable
```
allowed_values = [[] for i in range(...)]
allowed_values[MapOptIdx.nTiledDims] = [i + 1 for i in range(6)]
```

Reply:
Will work on that then.

Reply:
Done. What do you think?